staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go

```diff
@@ -432,11 +432,8 @@ func (e *Store) deleteWithoutFinalizers(ctx context.Context, name, key string, o
 		// requests to remove all finalizers from the object, so we
 		// ignore the NotFound error.
 		if storage.IsNotFound(err) {
-			_, err := e.finalizeDelete(ctx, obj, true)
```

Comment:
I don't think dropping the call to finalizeDelete is more correct. it means the returned object will not have been passed through any of the post-deletion methods (AfterDelete, Decorate, etc)

Reply:
+1

Also, this line needs to behave the same as https://github.com/kubernetes/kubernetes/blob/master/staging/src/k8s.io/apiserver/pkg/registry/generic/registry/store.go#L956-L963.